src/Goteo/Model/Faq.php

```diff
@@ -10,81 +10,171 @@
                 $fields,
                 faq.order as `order`
             FROM faq
             $joins
-            WHERE faq.id = :id
-            ", array(':id' => $id));
+            ";
+
+
+        if(is_string($id)) {
+            $sql .= "WHERE faq.slug = :slug";
+            $values = [':slug' => $id];
+        } else {
+            $sql .= "WHERE faq.id = :id";
+            $values = [':id' => $id];
+        }
+
+         //die(\sqldbg($sql, $values));
+        $query = static::query($sql, $values);
+
         $faq = $query->fetchObject(__CLASS__);
 
+        if (!$faq)
+            throw new ModelNotFoundException();
+
         return $faq;
     }
 
     /*
-     * Lista de proyectos destacados
+     * Lista de faqs
      */
-    public static function getAll ($section = 'node', $lang = null) {
+    public static function getAll ($subsection = 'node', $lang = null) {
         if(!$lang) $lang = Lang::current();
-        $values = array(':section' => $section);
+        $values = array(':subsection' => $subsection);
 
         list($fields, $joins) = self::getLangsSQLJoins($lang, Config::get('sql_lang'));
 
         $sql="SELECT
                     faq.id as id,
                     faq.node as node,
-                    faq.section as section,
+                    faq.subsection_id,
                     $fields,
                     faq.order as `order`
                 FROM faq
                 $joins
-                WHERE faq.section = :section
+                WHERE faq.subsection_id = :subsection
                 ORDER BY `order` ASC";
 
         $query = static::query($sql, $values);
 
         return $query->fetchAll(\PDO::FETCH_CLASS, __CLASS__);
     }
 
+    static public function getList(array $filters = [], int $offset = 0, int $limit = 10, bool $count = false, string $lang = null)
+    {
+        if(!$lang) $lang = Lang::current();
+        list($fields, $joins) = self::getLangsSQLJoins($lang, Config::get('sql_lang'));
+
+        $filter = [];
+        $values = [];
+
+        if ($filters['section']) {
+
+            // get subsections from a section
+            $subsections= FaqSubsection::getList(['section'=>$filters['section']]);
+
+            foreach ($subsections as $subsection)
+                $subsections_id[]=$subsection->id;
+
+            $filter[] = "faq.subsection_id in ('".implode("','", $subsections_id)."')";
+
+        }
+
+        if ($filters['subsection']) {
+            $filter[] = "faq.subsection_id = :subsection_id";
+            $values[':subsection_id'] = $filters['subsection'];
+
+        }
+
+        if($filter) {
+            $sql = " WHERE " . implode(' AND ', $filter);
+        }
+
+        if ($count) {
+            $sql = "
+                SELECT
+                    count(faq.id)
+                FROM faq
+                $joins
+                $sql
+            ";
+            $query = static::query($sql, $values);
+            return $query->fetchColumn();
+        }
```

Comment:
Could we remove the `$count` param from the `getList()` method, and create a `getListCount()` method, for example, so `getList()` is more concise?